<!DOCTYPE html>
<html lang="en"><head>
<script src="slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="slides_files/libs/quarto-html/popper.min.js"></script>
<script src="slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="slides_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="slides_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="slides_files/libs/quarto-contrib/iconify-2.1.0/iconify-icon.min.js"></script><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.57">

  <meta name="author" content="David Selby">
  <title>Knowledge-enhanced biomarker discovery</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="slides_files/libs/revealjs/dist/theme/quarto.css">
  <link href="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  <meta name="mermaid-theme" content="forest">
  <script src="slides_files/libs/quarto-diagram/mermaid.min.js"></script>
  <script src="slides_files/libs/quarto-diagram/mermaid-init.js"></script>
  <link href="slides_files/libs/quarto-diagram/mermaid.css" rel="stylesheet">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section data-background-color="black" data-background-image="images/burg_trifels2.jpg" data-background-opacity="0.5" data-background-size="contain" class="quarto-title-block center">
  <h1 class="title">Knowledge-enhanced biomarker discovery</h1>
  <p class="subtitle">Trifels Spring School 2025: AI in Bioinformatics</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
David Selby 
</div>
        <p class="quarto-title-affiliation">
            DFKI
          </p>
    </div>
</div>

  <p class="date">24th March 2025</p>
</section><section id="TOC">
<nav role="doc-toc"> 
<h2 id="toc-title">Agenda</h2>
<ul>
<li><span class="header-section-number">1</span> Introduction &amp; Motivation</li>
<li><span class="header-section-number">2</span> Prior knowledge</li>
<li><span class="header-section-number">3</span> Multi-omics integration</li>
<li><span class="header-section-number">4</span> Visible neural networks</li>
<li><span class="header-section-number">5</span> Hands-on</li>
<li><span class="header-section-number">6</span> Discussion</li>
</ul>
</nav>
</section>
<section>
<section class="title-slide slide level1 unnumbered unlisted center" data-background-image="images/annweiler.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5">
<h1>Preamble</h1>
<div class="cell">
<style type="text/css">
/* Style the table of contents */
#TOC ul li {
    list-style-type: none;
}
#TOC a {
    font-size: larger;
    color: black;
    font-weight: bold;
}
</style>
</div>
</section>
<section class="slide level2 unnumbered">
<h2>David Selby</h2>
<div class="columns">
<div class="column">
<p><img data-src="images/david.jpg"></p>
<p><strong>Senior Researcher</strong> <iconify-icon role="img" inline="" icon="twemoji:flag-germany" aria-label="Icon flag-germany from twemoji Iconify.design set." title="Germany"></iconify-icon> <iconify-icon role="img" inline="" icon="twemoji:flag-united-kingdom" aria-label="Icon flag-united-kingdom from twemoji Iconify.design set." title="United Kingdom"></iconify-icon><br>
Data Science &amp; its Applications<br>
<a href="mailto:david.selby@dfki.de">david.selby@dfki.de</a></p>
</div><div class="column">
<p><img data-src="images/dfki_logo_full.png"> <img data-src="images/rptu_logo.png"></p>
</div></div>
</section>
<section class="slide level2 unnumbered" data-background-image="images/annweiler.jpg" data-background-size="cover" data-background-color="black">
<h2></h2>
<div class="footer">
<p>© David Selby</p>
</div>
</section>
<section class="slide level2 unnumbered" data-background-image="images/burg_trifels1.jpg" data-background-size="cover" data-background-color="black">
<h2></h2>
<div class="footer">
<p>© David Selby</p>
</div>
</section>
<section class="slide level2 unnumbered" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black">
<h2></h2>
<div class="footer">
<p>© David Selby</p>
</div>
</section>
<section class="slide level2 unnumbered" data-background-image="images/burg_aussicht.jpg" data-background-size="cover" data-background-color="black">
<h2></h2>
<div class="footer">
<p>© David Selby</p>
</div>
</section>
<section class="slide level2 unnumbered">
<h2>Workshop objectives <iconify-icon role="img" inline="" icon="pajamas:issue-type-objective" aria-label="Icon issue-type-objective from pajamas Iconify.design set." title="Icon issue-type-objective from pajamas Iconify.design set."></iconify-icon></h2>
<p>By the end of this session, we aim to:</p>
<ul>
<li><iconify-icon role="img" inline="" icon="twemoji:brain" aria-label="Icon brain from twemoji Iconify.design set." title="Icon brain from twemoji Iconify.design set."></iconify-icon> <strong>Understand</strong> role of prior knowledge in biomarker discovery</li>
<li><iconify-icon role="img" inline="" icon="twemoji:dna" aria-label="Icon dna from twemoji Iconify.design set." title="Icon dna from twemoji Iconify.design set."></iconify-icon> <strong>Learn</strong> how to integrate biological context into workflows</li>
<li><iconify-icon role="img" inline="" icon="twemoji:hammer-and-wrench" aria-label="Icon hammer-and-wrench from twemoji Iconify.design set." title="Icon hammer-and-wrench from twemoji Iconify.design set."></iconify-icon> <strong>Explore</strong> tools for knowledge-guided analysis</li>
<li><iconify-icon role="img" inline="" icon="twemoji:construction" aria-label="Icon construction from twemoji Iconify.design set." title="Icon construction from twemoji Iconify.design set."></iconify-icon> <strong>Discuss</strong> challenges in knowledge-guided AI for biomedicine</li>
</ul>
</section></section>
<section>
<section class="title-slide slide level1 smaller center" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5" data-number="1">
<h1><span class="header-section-number">1</span> Introduction &amp; Motivation</h1>

</section>
<section class="slide level2 center unnumbered">
<h2></h2>
<p><span style="color: rgb(029, 058, 143);"><strong>Knowledge enhanced</strong></span> <span style="color: rgb(236, 097, 159);">(multi-omics)</span> <span style="color: rgb(106, 191, 163);"><strong>biomarker discovery</strong></span></p>
<div class="fragment">
<div>

</div>
<div class="columns quarto-layout-panel" data-layout="[0.3, 0.3, 0.3]">
<div class="quarto-layout-row">
<div class="column quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<p><iconify-icon role="img" inline="" icon="twemoji:brain" aria-label="Icon brain from twemoji Iconify.design set." title="Icon brain from twemoji Iconify.design set."></iconify-icon></p>
<p>Why? Where? How?</p>
</div><div class="column quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<p><iconify-icon role="img" inline="" icon="twemoji:abacus" aria-label="Icon abacus from twemoji Iconify.design set." title="Icon abacus from twemoji Iconify.design set."></iconify-icon></p>
<p>What makes it hard?</p>
</div><div class="column quarto-layout-cell" style="flex-basis: 33.3%;justify-content: flex-start;">
<p><iconify-icon role="img" inline="" icon="twemoji:dna" aria-label="Icon dna from twemoji Iconify.design set." title="Icon dna from twemoji Iconify.design set."></iconify-icon></p>
<p>What are we looking for?</p>
</div></div>
</div>
</div>
</section>
<section class="slide level2">

<p><img data-src="https://curatime.org/assets/610bc0d76b2b3a8ddbeeb5df_curatime_logo.svg" style="width:40.0%"></p>
<p><img data-src="https://curatime.org/img/containers/assets/curatime_unsere-losung_transparent.png/79c9599f71a662d02213fc803b87d53f.webp"></p>
</section>
<section class="slide level2">

<div class="cell" data-reveal="true" data-layout-align="default">
<div class="cell-output-display">
<div id="biomarker-history">
<div>
<pre class="mermaid mermaid-js" data-label="biomarker-history">timeline
    title A short history of biomarker discovery

%%    section Before AI Integration
      1949 : Introduction of 'biochemical markers' (Mundkur)
      1970s : Term 'biomarker' coined (1973) : Prostate-Specific Antigen discovered (1979)
      1990s : BRCA1 gene discovered by Mary-Claire King (1990) : BRCA1 gene cloned (1994)
%%    section After AI Integration
      2000s : 'biomarker' formally defined* (2000) : NGS introduced (2005) : QRISK (2007)
      2010s : liquid biopsy : endotypes : CRISPR : ImageNet : GANs : BERT : SHAP
      2020s : AlphaFold : GPT-3 : Blood-based biomarkers for Alzheimer's
</pre>
</div>
</div>
</div>
</div>

<div class="footer">
<p>Bodaghi, Fattahi &amp; Ramazani. Biomarkers. <em>Heliyon</em> (2023). doi:<a href="https://doi.org/10.1016/j.heliyon.2023.e13323">10.1016/j.heliyon.2023.e13323</a></p>
<p>Ng, S., Masarone, S., Watson, D. et al.&nbsp;The benefits and pitfalls of machine learning for biomarker discovery. <em>Cell Tissue Res</em> (2023). doi:<a href="https://doi.org/10.1007/s00441-023-03816-z">10.1007/s00441-023-03816-z</a></p>
</div>
<aside><div>
<p>*<strong>Definition</strong>: “indicators of biologic/pathogenic processes/responses … frequently measured and evaluated”</p>
</div></aside></section>
<section class="slide level2">

<h3>Key challenges</h3>
<ul>
<li>High dimensionality &amp; small sample sizes (<span class="math inline">\(p \gg n\)</span>)</li>
<li>Heterogeneous data modalities</li>
<li>Complexity–interpretability tradeoff</li>
<li>Validation in diverse cohorts</li>
<li>FAIRness of data, methods and tools</li>
</ul>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Signatures or biomarkers</strong></p>
</div>
<div class="callout-content">
<p>(Sets of) features predictive of a biological outcome</p>
</div>
</div>
</div>
<aside class="notes">
<ul>
<li>Dimensionality reduction</li>
<li>Integrating different molecular data, as well as non-molecular clinical data and imaging</li>
<li>Explainability, interpretability</li>
<li>Heavy reliance on TCGA and common tools may not generalize</li>
<li>FAIR = Findable, Accessible, Interoperable, Reusable</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">
<h2>Multi-omics biomarker discovery</h2>
<div class="columns">
<div class="column">
<ul>
<li>Contrasts with <strong>single-omics</strong></li>
<li>Combines
<ul>
<li>genomics, transcriptomics, proteomics, metabolomics, …</li>
<li>cliniomics, radiomics, …</li>
</ul></li>
<li>Identify robust <strong>signatures</strong> for disease diagnosis, prognosis or treatment</li>
</ul>
</div><div class="column">
<p><img data-src="images/multiomics.png"></p>
</div></div>
</section>
<section class="slide level2">
<h2>Knowledge-intensive machine learning</h2>
<ul>
<li><strong>Prior knowledge</strong> can guide model training</li>
<li><strong>Interpretability</strong> is crucial for clinical adoption</li>
<li>Easy to <strong>overfit</strong> in high-dimensional space</li>
<li>Biomedical analytical insights not easy to <strong>reproduce</strong>
<ul>
<li>data wrangling</li>
<li>domain expertise</li>
<li>model interpretations</li>
</ul></li>
</ul>
</section></section>
<section>
<section class="title-slide slide level1 smaller center" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5" data-number="2">
<h1><span class="header-section-number">2</span> Prior knowledge</h1>

</section>
<section class="slide level2">
<h2>What is prior knowledge?</h2>
<ul>
<li>scientific publications in literature</li>
<li>open datasets (e.g.&nbsp;TCGA, OpenML, UCI)</li>
<li>domain-specific databases (e.g.&nbsp;KEGG, Reactome, GO)</li>
<li>networks data (e.g.&nbsp;protein-protein interactions)</li>
<li>ontologies</li>
<li>expert knowledge (Bayesian decision-making)</li>
</ul>
<div class="fragment">
<p>How can prior knowledge be encoded in a <strong>transparent, reproducible</strong> way?</p>
</div>
</section>
<section class="slide level2">
<h2>Approaches</h2>
<ol type="1">
<li>Regularization</li>
<li>Knowledge graphs</li>
<li>Biologically-informed architectures</li>
</ol>
</section>
<section class="slide level2 smaller">
<h2>Information theory</h2>
<p><strong>Entropy</strong> <span class="math inline">\(H(X)\)</span> quantifies uncertainty in a random variable <span class="math inline">\(X\)</span></p>
<ul>
<li>Higher entropy → more uncertainty</li>
<li>Lower entropy → more certainty</li>
</ul>
<p><strong>Prior knowledge</strong> reduces entropy:</p>
<ul>
<li>Narrows the hypothesis space</li>
<li>Focuses on biologically plausible solutions</li>
</ul>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Example: biomarker discovery</strong></p>
</div>
<div class="callout-content">
<p><strong>Without</strong> prior knowledge, search space includes <strong>all possible gene combinations</strong> (high entropy, expensive).</p>
<p><strong>With</strong> prior knowledge, focus on <strong>pathways, interactions, or known gene sets</strong> (low entropy, faster convergence).</p>
</div>
</div>
</div>
</section>
<section class="slide level2 unnumbered smaller">
<h2></h2>
<p><strong>Mutual information</strong> <span class="math inline">\(I(X;Y)\)</span> measures shared information between <span class="math inline">\(X\)</span> (data) and <span class="math inline">\(Y\)</span> (prior knowledge).</p>
<p>Higher <span class="math inline">\(I(X;Y)\)</span> → more effective integration of prior knowledge</p>
<p>Mutual information can be expressed <span class="math display">\[
I(X;K) = H(X) - H(X|K),
\]</span> where</p>
<ul>
<li><span class="math inline">\(H(X)\)</span> is the entropy (uncertainty) of the data <span class="math inline">\(X\)</span></li>
<li><span class="math inline">\(H(X|K)\)</span> is the conditional entropy of <span class="math inline">\(X\)</span> given prior knowledge <span class="math inline">\(K\)</span>.</li>
</ul>
<p>Reduction in entropy is equivalent to the mutual information: <span class="math inline">\(\Delta H = I(X;K).\)</span></p>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tip</strong></p>
</div>
<div class="callout-content">
<p>The greater the mutual information <span class="math inline">\(I(X;K)\)</span>, the more effective the prior knowledge <span class="math inline">\(K\)</span> is in narrowing the hypothesis space.</p>
</div>
</div>
</div>
<div class="footer">
<p>Shannon, C.E. A Mathematical Theory of Communication. <em>Bell System Technical Journal</em> (1948). doi:<a href="https://doi.org/10.1002/j.1538-7305.1948.tb01338.x">10.1002/j.1538-7305.1948.tb01338.x</a></p>
</div>
</section>
<section class="slide level2">


<img data-src="https://upload.wikimedia.org/wikipedia/commons/d/d4/Entropy-mutual-information-relative-entropy-relation-diagram.svg" class="r-stretch quarto-figure-center"><p class="caption">Entropy <span class="math inline">\(H(\cdot)\)</span>, conditional entropy <span class="math inline">\(H(\cdot\mid\cdot)\)</span> and mutual information <span class="math inline">\(I(\cdot; \cdot)\)</span></p></section>
<section class="slide level2">

<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Example of entropy in biomarker discovery</strong></p>
</div>
<div class="callout-content">
<p><strong>Without Prior Knowledge</strong>:</p>
<ul>
<li>Searching among <strong>10,000 genes</strong>.</li>
<li>Entropy: <span class="math inline">\(H(X) = \log_2(10,000) \approx 13.29 \text{ bits.}\)</span></li>
</ul>
<p><strong>With Prior Knowledge</strong>:</p>
<ul>
<li>Prior knowledge narrows the search to <strong>100 candidate genes</strong>.</li>
<li>Entropy: <span class="math inline">\(H(X|K) = \log_2(100) \approx 6.64 \text{ bits.}\)</span></li>
</ul>
<p><strong>Mutual Information</strong></p>
<ul>
<li>Reduction in entropy:<br>
<span class="math inline">\(I(X;K) = H(X) - H(X|K) = 13.29 - 6.64 = 6.65 \text{ bits.}\)</span></li>
</ul>
</div>
</div>
</div>
</section>
<section class="slide level2 unnumbered center">
<h2></h2>
<div class="callout callout-warning callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Lower entropy ≠ correctness</strong></p>
</div>
<div class="callout-content">
<p>Prior knowledge (and data) can still be <strong>wrong/biassed</strong>!</p>
<p>Then <em>smaller</em> hypothesis space → barking up the <em>wrong</em> tree.</p>
</div>
</div>
</div>
</section></section>
<section>
<section class="title-slide slide level1 smaller center" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5" data-number="3">
<h1><span class="header-section-number">3</span> Multi-omics integration</h1>

</section>
<section class="slide level2 unnumbered" data-background-image="https://www.researchgate.net/publication/346929297/figure/fig1/AS:11431281241521883@1715131118932/The-complexity-of-multi-omics-merger-of-omics-driven-biology-data-science-informatics.tif" data-background-size="contain">
<h2></h2>
<div class="footer">
<p>Krassowski at al.&nbsp;(2020). State of the Field in Multi-Omics Research. <em>Frontiers in Genetics.</em> doi:<a href="https://doi.org/10.3389/fgene.2020.610798">10.3389/fgene.2020.610798</a></p>
</div>
</section>
<section class="slide level2">
<h2>Single-omics methods</h2>
<table class="caption-top">
<colgroup>
<col style="width: 43%">
<col style="width: 56%">
</colgroup>
<thead>
<tr class="header">
<th>Task</th>
<th>Tools</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><iconify-icon role="img" inline="" icon="twemoji:bar-chart" aria-label="Icon bar-chart from twemoji Iconify.design set." title="Icon bar-chart from twemoji Iconify.design set."></iconify-icon> Differential expression analysis</td>
<td>DESeq2, edgeR, limma</td>
</tr>
<tr class="even">
<td><iconify-icon role="img" inline="" icon="vaadin:cluster" aria-label="Icon cluster from vaadin Iconify.design set." title="Icon cluster from vaadin Iconify.design set."></iconify-icon> Clustering samples/features</td>
<td><span class="math inline">\(k\)</span>-means, hierarchical clustering, <span class="math inline">\(t\)</span>-SNE, UMAP</td>
</tr>
<tr class="odd">
<td><iconify-icon role="img" inline="" icon="twemoji:dna" aria-label="Icon dna from twemoji Iconify.design set." title="Icon dna from twemoji Iconify.design set."></iconify-icon> Pathway enrichment analysis</td>
<td>GSEA, Reactome</td>
</tr>
<tr class="even">
<td><iconify-icon role="img" inline="" icon="twemoji:chart-increasing" aria-label="Icon chart-increasing from twemoji Iconify.design set." title="Icon chart-increasing from twemoji Iconify.design set."></iconify-icon> Predictive modelling</td>
<td>GLMs, random forests, SVM, XGBoost, NNs</td>
</tr>
</tbody>
</table>
</section>
<section class="slide level2 smaller">
<h2>Single-omics challenges</h2>
<table class="caption-top">
<colgroup>
<col style="width: 25%">
<col style="width: 40%">
<col style="width: 34%">
</colgroup>
<thead>
<tr class="header">
<th>Challenge</th>
<th>Methods</th>
<th>Tools/Packages</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(p \gg n\)</span></td>
<td>Dimensionality reduction</td>
<td>PCA, t-SNE, UMAP, LASSO, ElasticNet</td>
</tr>
<tr class="even">
<td>Batch effects</td>
<td>Batch effect correction</td>
<td>ComBat (sva), limma, Harmony</td>
</tr>
<tr class="odd">
<td>Missing data</td>
<td>Imputation</td>
<td>MICE, KNN imputation, MissForest</td>
</tr>
</tbody>
</table>
<div class="fragment">
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tip</strong></p>
</div>
<div class="callout-content">
<p>Single-omics data is <strong>tabular</strong>, for which <strong>tree-based models</strong> (e.g.&nbsp;random forests) can outperform deep learning.</p>
</div>
</div>
</div>
<div class="footer">
<p>Grinszjtajn et al.&nbsp;<a href="https://proceedings.neurips.cc/paper_files/paper/2022/hash/0378c7692da36807bdec87ab043cdadc-Abstract-Datasets_and_Benchmarks.html">Why do tree-based models still outperform deep learning on typical tabular data?</a> <em>NeurIPS.</em> (2022)</p>
</div>
</div>
</section>
<section class="slide level2">

<div class="callout callout-warning callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Multi-omics data bring extra challenges</strong></p>
</div>
<div class="callout-content">
<ul>
<li><strong>Integration</strong> of different data types</li>
<li><strong>Interpretation</strong> of complex interactions</li>
</ul>
</div>
</div>
</div>
</section>
<section class="slide level2 center">
<h2>Multi-omics datasets</h2>
<div class="columns">
<div class="column">
<p><img data-src="images/multiomics.png"></p>
</div><div class="column">
<ul>
<li>Tabular data</li>
<li>High-dimensional</li>
<li>Small samples</li>
<li><strong>Multimodal structure</strong></li>
</ul>
</div></div>
</section>
<section class="slide level2">
<h2>Why deep learning?</h2>
<p>Why not just use classical methods?</p>
<h3>Classical approaches</h3>
<dl>
<dt>Multi-omics factor analysis (MOFA)</dt>
<dd>
<p>Unsupervised, generalization of PCA</p>
</dd>
<dt>Canonical correlation analysis (CCA)</dt>
<dd>
<p>Find linear combinations of features that are maximally correlated</p>
</dd>
<dt>Grouped LASSO</dt>
<dd>
<p>(Linearly) penalize groups of features together</p>
</dd>
</dl>
</section>
<section class="slide level2">
<h2>Why deep learning?</h2>
<p>Why not just use classical methods? Isn’t it tabular data?</p>
<div>
<ul>
<li class="fragment"><strong>Non-linear</strong> relationships</li>
<li class="fragment"><strong>Interactions</strong> between features</li>
<li class="fragment"><strong>Representation learning</strong> from raw data</li>
<li class="fragment"><strong>End-to-end</strong> learning</li>
</ul>
</div>
<div class="fragment">
<div class="cell" data-reveal="true" data-layout-align="default">
<div class="cell-output-display">
<div id="classical-vs-dl-pipeline">
<div>
<pre class="mermaid mermaid-js" data-label="classical-vs-dl-pipeline">flowchart LR
    A[Feature Selection] --&gt; B[Modelling]
    B --&gt; C[Pathway Enrichment]
</pre>
</div>
</div>
</div>
</div>
<div class="footer">
<p>Park et al.&nbsp;Sparse overlapping group lasso for integrative multi-omics analysis. <em>J Comput Biol.</em> (2015) doi:<a href="https://doi.org/10.1089/cmb.2014.0197">10.1089/cmb.2014.0197</a></p>
</div>
</div>
</section>
<section class="slide level2">
<h2>Deep learning architectures</h2>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Question</strong></p>
</div>
<div class="callout-content">
<p>Which neural network architectures are suitable for omics?</p>
</div>
</div>
</div>
</section>
<section class="slide level2 smaller">
<h2>Multimodal fusion</h2>
<p>When should we combine omics layers?</p>

<img data-src="https://cdn.ncbi.nlm.nih.gov/pmc/blobs/f809/8829812/7f899d7be8ff/gr2.jpg" class="r-stretch"><div class="footer">
<p>Cai, Poulos, Liu &amp; Zhong. Machine learning for multi-omics data integration in cancer. <em>iScience.</em> (2022). doi:<a href="https://doi.org/10.1016/j.isci.2022.103798">10.1016/j.isci.2022.103798</a></p>
</div>
</section>
<section class="slide level2">
<h2>Multimodal fusion</h2>
<p>When should we combine omics layers?</p>
<dl>
<dt>Early</dt>
<dd>
<p>easier, loss of information, worse performance*</p>
</dd>
<dt>Intermediate (mixed, joint)</dt>
<dd>
<p>modality-specific layers, but harder to train</p>
</dd>
<dt>Late</dt>
<dd>
<p>may not capture interactions</p>
</dd>
</dl>
<div class="footer">
<p>*Hauptmann, T., Kramer, S. A fair experimental comparison of neural network architectures for latent representations of multi-omics for drug response prediction.&nbsp;<em>BMC Bioinformatics</em>&nbsp;24, 45 (2023). doi:<a href="https://doi.org/10.1186/s12859-023-05166-7">10.1186/s12859-023-05166-7</a></p>
</div>
</section>
<section class="slide level2">
<h2>Late fusion (MOLI)</h2>

<img data-src="https://cdn.ncbi.nlm.nih.gov/pmc/blobs/3f5e/6612815/2f27faef728d/btz318f1.jpg" class="r-stretch"><div class="footer">
<p>Sharifi-Noghabi et al.&nbsp;<em>Bioinformatics</em>. 2019. doi: <a href="https://doi.org/10.1093/bioinformatics/btz318">10.1093/bioinformatics/btz318</a></p>
</div>
<aside class="notes">
<p>Hello, world</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">
<h2>Graph neural networks (GNNs)</h2>

<img data-src="https://www.mdpi.com/ijms/ijms-25-02788/article_deploy/html/images/ijms-25-02788-g001.png" class="r-stretch quarto-figure-center"><p class="caption">MOGAT</p><div class="footer">
<p>Tanvir et al.&nbsp;MOGAT. <em>International Journal of Molecular Sciences</em>, (2024). doi:<a href="https://doi.org/10.3390/ijms25052788">10.3390/ijms25052788</a></p>
</div>
</section>
<section class="slide level2">
<h2>Model explanations</h2>

<img data-src="images/nn.png" class="r-stretch quarto-figure-center"><p class="caption">Can we call hidden nodes in neural networks ‘biomarkers’?</p></section>
<section class="slide level2">
<h2>Model explanations</h2>
<p>Input-level explanations:</p>
<ul>
<li><span class="math inline">\(p\)</span>-values, features importance</li>
<li>DeepLIFT</li>
<li>SHAP</li>
<li>LIME</li>
</ul>
<p>→ <em>post-hoc</em> gene-set enrichment analysis (GSEA) or “pathway analysis”</p>
<p><img data-src="https://reactome.org/templates/favourite/images/logo/logo.png"> &nbsp; <!--![](https://upload.wikimedia.org/wikipedia/en/8/80/KEGG_database_logo.gif)--></p>
</section>
<section class="slide level2 smaller">
<h2>Gene set enrichment analysis</h2>
<ol type="1">
<li><p>Set of genes <span class="math inline">\(G = \{g_1, g_2, \dots, g_N\}\)</span>. Order by ranking metric <span class="math inline">\(S(g_i)\)</span> (e.g.&nbsp;<em>t</em>-statistic)</p></li>
<li><p>Compute <strong>enrichment score</strong> using running sum statistics, or <strong>overrepresentation score</strong> with hypergeometric test: <span class="math display">\[
P(X = x) = \frac{\binom{M}{x} \binom{N-M}{n-x}}{\binom{N}{n}}
\]</span> with <span class="math inline">\(p\)</span>-value <span class="math display">\[
p = \sum_{i=x}^{\min(M, K)} P(X = i).
\]</span></p></li>
</ol>
</section>
<section class="slide level2">
<h2>What is a pathway (gene set)?</h2>

<img data-src="https://reactome.org/images/baac009f1.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Reactome pathways</p><p>A pathway is a set of genes that are known to interact in a biological process.</p>
<div class="footer">
<p><a href="https://reactome.org/" class="uri">https://reactome.org/</a></p>
</div>
</section>
<section class="slide level2">
<h2>Pathway databases</h2>
<div class="columns">
<div class="column">
<p><img data-src="https://reactome.org/templates/favourite/images/logo/logo.png"></p>
</div><div class="column">
<p><img data-src="https://upload.wikimedia.org/wikipedia/en/8/80/KEGG_database_logo.gif"></p>
</div></div>
<p><img data-src="https://geneontology.org/assets/go-logo.png"></p>
</section>
<section class="slide level2 centered">
<h2></h2>
<section class="callout-info">
<h3>Pathway enrichment analysis</h3>
<p>What are the disadvantages of using <em>post-hoc</em> GSEA or GSOA?</p>
</section>
</section></section>
<section>
<section class="title-slide slide level1 smaller center" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5" data-number="4">
<h1><span class="header-section-number">4</span> Visible neural networks</h1>

</section>
<section class="slide level2">
<h2>Feedforward neural network</h2>

<img data-src="images/nn.png" class="r-stretch quarto-figure-center"><p class="caption"><iconify-icon role="img" inline="" icon="twemoji:magnifying-glass-tilted-left" aria-label="Icon magnifying-glass-tilted-left from twemoji Iconify.design set." title="Icon magnifying-glass-tilted-left from twemoji Iconify.design set."></iconify-icon> Where are the biomarkers?</p></section>
<section class="slide level2">
<h2>Visible neural network (VNN)</h2>

<img data-src="https://ars.els-cdn.com/content/image/1-s2.0-S0092867418307190-gr2_lrg.jpg" class="r-stretch quarto-figure-center"><p class="caption">DCell</p><div class="footer">
<p>Yu et al.&nbsp;Visible Machine Learning for Biomedicine. <em>Cell</em> (2018) doi:<a href="https://doi.org/10.1016/j.cell.2018.05.056">10.1016/j.cell.2018.05.056</a></p>
</div>
</section>
<section class="slide level2">
<h2>Knowledge-primed neural network</h2>

<img data-src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs13059-020-02100-5/MediaObjects/13059_2020_2100_Fig1_HTML.png" class="r-stretch"><div class="footer">
<p>Fortelny &amp; Bock. Knowledge-primed neural networks. <em>Genome Biology</em> (2020). doi:<a href="https://doi.org/10.1186/s13059-020-02100-5">10.1186/s13059-020-02100-5</a></p>
</div>
</section>
<section class="slide level2">
<h2>Biologically informed neural network (BINN)</h2>

<img data-src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41586-021-03922-4/MediaObjects/41586_2021_3922_Fig1_HTML.png" class="r-stretch quarto-figure-center"><p class="caption">P-Net</p><div class="footer">
<p>Elmarakeby et al.&nbsp;Biologically informed deep neural network for prostate cancer discovery. <em>Nature</em>, (2021). doi:<a href="https://doi.org/10.1038/s41586-021-03922-4">10.1038/s41586-021-03922-4</a></p>
</div>
</section>
<section class="slide level2">
<h2>VAE enhanced by gene annotations (VEGA)</h2>

<img data-src="https://vega-documentation.readthedocs.io/en/latest/_images/vega.png" class="r-stretch quarto-figure-center"><p class="caption">VEGA</p><div class="footer">
<p>Seninge, et al.&nbsp;VEGA. <em>Nature Communications</em> (2021). doi:<a href="https://doi.org/10.1038/s41467-021-26017-0">10.1038/s41467-021-26017-0)</a></p>
</div>
</section>
<section class="slide level2">
<h2>Anatomy of a BINN/VNN</h2>

<img data-src="images/binn_natrevgen.png" style="width:80.0%" class="r-stretch"><div class="footer">
<p>Selby, D.A. et al.&nbsp;<em>Nature Reviews Genetics</em> (2025). doi:<a href="https://doi.org/10.1038/s41576-025-00826-1">10.1038/s41576-025-00826-1</a></p>
</div>
</section>
<section class="slide level2">
<h2>How to build a VNN</h2>
<ol type="1">
<li>Start with dense sequential neural network</li>
<li>Use adjacency matrix of inputs → pathways as a <strong>masking matrix</strong></li>
<li>(Optional) for next layer, mask = mapping of pathways → higher pathways</li>
<li>Omit all masked weights from backpropagation</li>
</ol>
<div class="callout callout-warning callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Warning</strong></p>
</div>
<div class="callout-content">
<p>What if the pathways aren’t all the same depth?</p>
</div>
</div>
</div>
</section>
<section class="slide level2">
<h2>Pathway discovery with BINNs</h2>

<img data-src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12859-023-05535-2/MediaObjects/12859_2023_5535_Fig1_HTML.png" class="r-stretch quarto-figure-center"><p class="caption">PathExpSurv</p><div class="footer">
<p>Hou et al.&nbsp;PathExpSurv. <em>BMC Bioinformatics</em>, (2023). doi:<a href="https://doi.org/10.1186/s12859-023-05535-2">10.1186/s12859-023-05535-2</a></p>
</div>
</section>
<section class="slide level2">
<h2>Interpretable AutoML discovery</h2>
<div class="columns">
<div class="column">
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Open question</strong></p>
</div>
<div class="callout-content">
<p>If (1) VNNs perform better, and (2) Pathway DBs are incomplete; then</p>
<p><strong>Can neural architecture search over VNNs discover new pathways?</strong></p>
</div>
</div>
</div>
</div><div class="column">
<p><img data-src="https://towardsdatascience.com/wp-content/uploads/2020/09/1nicFUkeUpWMW1w_hUVtZiw.png"></p>
</div></div>
<div class="footer">
<p>Han et al.&nbsp;(2015). arXiv:<a href="https://arxiv.org/abs/1506.02626v3">1506.02626v3</a></p>
</div>
</section></section>
<section>
<section class="title-slide slide level1 smaller center" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5" data-number="5">
<h1><span class="header-section-number">5</span> Hands-on</h1>

</section>
<section class="slide level2 smaller">
<h2>Accessible tools for BINNs</h2>
<p>…would be a very good idea</p>
<div class="fragment">
<div>

</div>
<div class="columns quarto-layout-panel" data-layout="[0.5, -0.05, 0.5]">
<div class="quarto-layout-row">
<div class="column quarto-layout-cell" style="flex-basis: 47.6%;justify-content: flex-start;">
<h3><a href="https://github.com/InfectionMedicineProteomics/BINN">BINN</a></h3>
<ul>
<li><code>pip install binn</code></li>
<li>supervised learning</li>
<li>proteomics or single-omics</li>
<li>omics data: as <code>pandas</code> or csv</li>
<li>pathways: ‘Reactome’ or own csv (<code>parent,child</code>)</li>
</ul>
</div><div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.8%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="column quarto-layout-cell" style="flex-basis: 47.6%;justify-content: flex-start;">
<h3><a href="https://github.com/LucasESBS/vega">VEGA</a></h3>
<ul>
<li><code>pip install scvega</code></li>
<li>unsupervised learning</li>
<li>transcriptomics</li>
<li>omics data: as Anndata</li>
<li>pathways: GMT from <a href="https://www.gsea-msigdb.org/gsea/index.jsp">MSigDB</a></li>
</ul>
</div></div>
</div>
</div>
<div class="fragment">
<p>…and various other paper repos you can <code>git clone</code>.</p>
</div>
<div class="fragment">
<p>Nothing for R… <a href="https://torch.mlverse.org/">yet</a>!</p>
</div>
</section>
<section class="slide level2 center">
<h2></h2>
<div class="callout callout-warning callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>What’s missing?</strong></p>
</div>
<div class="callout-content">
<ul>
<li>multi-omics</li>
<li>integration with clinical data</li>
<li>survival analysis (censored time-to-event)</li>
<li>R + BioConductor</li>
</ul>
</div>
</div>
</div>
</section>
<section class="slide level2">
<h2>The <code>binn</code> package</h2>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Installation</strong></p>
</div>
<div class="callout-content">
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href=""></a><span class="ex">pip</span> install binn==0.1.1         <span class="co"># 0.1.0 has a bug</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Usage</strong></p>
</div>
<div class="callout-content">
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href=""></a><span class="im">from</span> binn <span class="im">import</span> BINN</span>
<span id="cb2-2"><a href=""></a>binn <span class="op">=</span> BINN(</span>
<span id="cb2-3"><a href=""></a>    data_matrix<span class="op">=</span>input_data,</span>
<span id="cb2-4"><a href=""></a>    network_source<span class="op">=</span><span class="st">"reactome"</span>,</span>
<span id="cb2-5"><a href=""></a>    input_source<span class="op">=</span><span class="st">"uniprot"</span>,</span>
<span id="cb2-6"><a href=""></a>    n_layers<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb2-7"><a href=""></a>    dropout<span class="op">=</span><span class="fl">0.2</span></span>
<span id="cb2-8"><a href=""></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Documentation</strong></p>
</div>
<div class="callout-content">
<p><a href="https://infectionmedicineproteomics.github.io/BINN/" class="uri">https://infectionmedicineproteomics.github.io/BINN/</a></p>
</div>
</div>
</div>
</section>
<section class="slide level2 smaller">
<h2>The binn package - inputs</h2>
<p>What’s required:</p>
<div>

</div>
<div class="columns quarto-layout-panel" data-layout="[0.3, -0.05, 0.3, -0.05, 0.3]">
<div class="quarto-layout-row">
<div class="column quarto-layout-cell" style="flex-basis: 30.0%;justify-content: flex-start;">
<p><strong><code>data_matrix</code></strong></p>
<ul>
<li><span class="math inline">\(n \times p\)</span> matrix of omics data</li>
</ul>
<p><strong><code>mapping</code></strong></p>
<ul>
<li><span class="math inline">\(p \times 2\)</span> 2 column matrix of features to pathways</li>
<li>or <code>'uniprot'</code></li>
</ul>
</div><div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.0%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="column quarto-layout-cell" style="flex-basis: 30.0%;justify-content: flex-start;">
<p><strong><code>pathways</code></strong></p>
<ul>
<li><span class="math inline">\(? \times 2\)</span> matrix: parent → child pathway</li>
<li>or <code>'reactome'</code></li>
</ul>
</div><div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 5.0%;justify-content: flex-start;">
<p>&nbsp;</p>
</div>
<div class="column quarto-layout-cell" style="flex-basis: 30.0%;justify-content: flex-start;">
<p><strong><code>design_matrix</code></strong></p>
<ul>
<li>matrix <span class="math inline">\(n \times 2\)</span> of targets <code>sample,group</code></li>
<li><code>group</code> = output class</li>
</ul>
</div></div>
</div>
<div class="footer">
<p><a href="https://infectionmedicineproteomics.github.io/BINN/" class="uri">https://infectionmedicineproteomics.github.io/BINN/</a></p>
</div>
</section>
<section class="slide level2">
<h2>The <code>binn</code> package - visualization</h2>

<img data-src="https://infectionmedicineproteomics.github.io/BINN/img/interpreted_binn.png" class="r-stretch quarto-figure-center"><p class="caption">Node importance plot</p><div class="footer">
<p><a href="https://infectionmedicineproteomics.github.io/BINN/" class="uri">https://infectionmedicineproteomics.github.io/BINN/</a></p>
</div>
</section>
<section class="slide level2">
<h2>Worksheet</h2>

<img data-src="images/qr_code.png" class="quarto-figure quarto-figure-center r-stretch"><p>Colab notebook and slides: <a href="https://github.com/datasciapps/trifels2025">github.com/datasciapps/trifels2025</a></p>
</section></section>
<section>
<section class="title-slide slide level1 smaller center" data-background-image="images/burg_trifels2.jpg" data-background-size="cover" data-background-color="black" data-background-opacity="0.5" data-number="6">
<h1><span class="header-section-number">6</span> Discussion</h1>

</section>
<section class="slide level2 smaller">
<h2>Challenges</h2>
<ul>
<li><strong>Lack of Robust Tools</strong>: Few standardized frameworks, poor interpretability and limited benchmarks.</li>
<li><strong>Alternative Models</strong>:
<ul>
<li><strong>Classical methods</strong>: faster, more interpretable, statistically robust</li>
<li><strong>GNNs</strong>: Better structural representation but need high-quality pathway graphs.<br>
</li>
<li><strong>Transformers/LLMs</strong>: Strong sequence modeling but lack explicit pathway structure.<br>
</li>
<li><strong>Hybrid Approaches</strong>: Combining VNNs with GNNs or transformers may improve learning.<br>
</li>
</ul></li>
<li><strong>Pathway Databases Are Incomplete</strong>: KEGG, Reactome, and GO suffer from curation bias, lack context specificity, and may be outdated.</li>
</ul>
</section>
<section class="slide level2 smaller">
<h2>Open questions</h2>
<h3>Biological fidelity</h3>
<ul>
<li>Is a BINN really a digital twin of a cell?</li>
<li>Biological sparsity or just noisy sparsity?</li>
</ul>
<h3>Model validation</h3>
<ul>
<li>Synthetic and experimental data for validation</li>
<li>Uncertainty-aware architectures to quantify confidence in biological predictions.</li>
</ul>
<h3>Performance vs interpretability</h3>
<ul>
<li>Mechanistic insight vs.&nbsp;predictive accuracy: which should take priority?</li>
<li>Do black-box models merely reinforce existing biases in pathway databases?</li>
</ul>
</section>
<section class="slide level2">
<h2>Things we didn’t cover today</h2>
<ul>
<li>Causal inference</li>
<li>Dynamic updating of knowledge graphs</li>
<li>Bayesian prior elicitation</li>
<li>GenAI: LLM agents &amp; retrieval-augmented generation</li>
</ul>
</section></section>
<section class="title-slide slide level1 unlisted unnumbered center">
<h1>Thank you!</h1>
<div class="columns">
<div class="column" style="width:70%;">
<p>&nbsp;</p>
<p><iconify-icon role="img" inline="" icon="twemoji:link" aria-label="Icon link from twemoji Iconify.design set." title="Icon link from twemoji Iconify.design set."></iconify-icon> <a href="https://github.com/datasciapps/trifels2025">github.com/datasciapps/trifels2025</a><br>
<iconify-icon role="img" inline="" icon="twemoji:envelope" aria-label="Icon envelope from twemoji Iconify.design set." title="Icon envelope from twemoji Iconify.design set."></iconify-icon> <a href="mailto:david.selby@dfki.de?Subject=Trifels%20Spring%20School">david.selby@dfki.de</a></p>
</div><div class="column" style="width:30%;">
<p><img data-src="images/qr_code.png"></p>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Further reading</strong></p>
</div>
<div class="callout-content">
<p>Selby, D.A. et al.&nbsp;Beyond the black box with biologically informed neural networks. <em>Nature Reviews Genetics</em> (2025). doi:<a href="https://doi.org/10.1038/s41576-025-00826-1">10.1038/s41576-025-00826-1</a></p>
</div>
</div>
</div>
<div class="footer">
<p>To export these slides, hit <code>E</code> and then <em>Print to PDF</em>.</p>
</div>
<div class="quarto-auto-generated-content">
<p><img src="images/dfki_black.png" class="slide-logo"></p>
<div class="footer footer-default">

</div>
</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="slides_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'fade',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>